src/hotspot/share/utilities/lockFreeQueue.inline.hpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_UTILITIES_LOCKFREEQUEUE_INLINE_HPP
+#define SHARE_UTILITIES_LOCKFREEQUEUE_INLINE_HPP
+
+#include "runtime/atomic.hpp"
+#include "runtime/thread.inline.hpp"
+#include "utilities/globalCounter.inline.hpp"
```

Comment:
Presumably this include is the reason for putting the `pop` support in an inline.hpp file.  But it seems clumsy to have most of the implementation in the ordinary header and this one function here, esp. since I suspect most clients will end up needing to include this file.  So I'm suggesting move all of the implementation here.